core/clients/6poc/web/views/mainView.jsx

```diff
@@ -0,0 +1,193 @@
+"use strict";
+
+var React = require('react');
+var $ = require('jquery');
+
+var MapView =  require('./mapView.jsx');
+
+module.exports = React.createClass({
+  getInitialState: function() {
+    var availableCategories = [
+      'aluminium',
+      'batteries',
+      'beverage_carton',
+      'bicycles',
+      'books',
+      'cans',
+      'cardboard',
+      'cds',
+      'chipboard',
+      'christmas_trees',
+      'clothes',
+      'computers',
+      'cooking_oil',
+      'cork',
+      'engine_oil',
+      'foil',
+      'furniture',
+      'glass',
+      'glass_bottles',
+      'green_waste',
+      'garden_waste',
+      'hardcode',
+      'hazardous_waste',
+      'light_bulbs',
+      'magazines',
+      'mobile_phones',
+      'music',
+      'newspaper',
+      'organic',
+      'paint',
+      'paper',
+      'paper_packaging',
+      'plastic',
+      'plastic_bags',
+      'plastic_bottles',
+      'plastic_packaging',
+      'polyester',
+      'printer_cartridges',
+      'rubble',
+      'scrap_metal',
+      'sheet_metal',
+      'shoes',
+      'tyres',
+      'waste',
+      'waste_oil',
+      'white_goods',
+      'wood']
+    return {result: null, mode: 0, condition: 0, radius: 50, availableCategories: availableCategories, categories: [], searchCpt: 0};
+  },
+  componentDidMount: function() {
+    var self = this;
+
+    var where = React.findDOMNode(this.refs.where);
+    google.maps.event.addDomListener(window, 'load', function() {
+      if (where) {
+        var autocomplete = new google.maps.places.Autocomplete(where, { types: ['geocode'] });
+        google.maps.event.addListener(autocomplete, 'place_changed', function(){
+          var place = autocomplete.getPlace();
+          var address = '';
+          if (place.address_components) {
+            address = [
+              (place.address_components[0] && place.address_components[0].short_name || ''),
+              (place.address_components[1] && place.address_components[1].short_name || ''),
+              (place.address_components[2] && place.address_components[2].short_name || '')
+            ].join(' ');
+          }
+          self.setState({geoloc: {lat: place.geometry.location.G, lon: place.geometry.location.K}, placeName: address})
+        });
+      }
+    })
+  },
+  launchSearch: function(){
+    var self = this;
+
+    var options =  React.findDOMNode(this.refs.what).options;
+    var categories = [];
+    for(var i=0; i<options.length; ++i){
+      if(options[i].selected == false) continue;
+      var value = options[i].value;
+      categories.push(value);
+    }
+    var data = {
+      'placeName': this.state.placeName,
+      'radius': this.state.radius,
+      'categories': categories,
+      'geoloc': this.state.geoloc
+    };
+    $.ajax({
```

Comment:
At least, this belongs to its own file, so concerns of fetching data and displaying them are separated (the scheduling via `launchSearch` may stay here for now)